python/ray/experimental/serve/queues.py

```diff
@@ -1,43 +1,69 @@
-from collections import defaultdict, deque
+import asyncio
+import copy
+from collections import defaultdict
+import time
+from typing import DefaultDict, Union, List
+import pickle
 
-import numpy as np
+# Note on choosing blist instead of stdlib heapq
+# 1. pop operation should be O(1) (amortized)
+#    (helpful even for batched pop)
+# 2. There should not be significant overhead in
+#    maintaining the sorted list.
+# 3. The blist implementation is fast and uses C extensions.
+import blist
 
 import ray
 from ray.experimental.serve.utils import logger
-import itertools
-from blist import sortedlist
-import time
 
 
 class Query:
-    def __init__(self,
-                 request_args,
-                 request_kwargs,
-                 request_context,
-                 request_slo_ms,
-                 result_object_id=None):
+    def __init__(self, request_args, request_kwargs, request_context,
+                 request_slo_ms):
         self.request_args = request_args
         self.request_kwargs = request_kwargs
         self.request_context = request_context
 
-        if result_object_id is None:
-            self.result_object_id = ray.ObjectID.from_random()
-        else:
-            self.result_object_id = result_object_id
+        self.async_future = asyncio.get_event_loop().create_future()
 
         # Service level objective in milliseconds. This is expected to be the
         # absolute time since unix epoch.
         self.request_slo_ms = request_slo_ms
 
+    def ray_serialize(self):
```

Comment:
Can you add comment on why the serialiser are needed? 
- The async future satisfied by the query stays insider the router
- task runner just returns the result for one or more queries